### ISSUE TITLE
Build fails on linux due to libusb

### ISSUE THREAD

**Issue by MGThePro** [yuzu-emu/yuzu#4279]
The current version (77bdd09) fails to build on linux due to libusb.h missing
I'm on arch linux, kernel 5.7.4-8-tkg-pds and gcc 10.1.0-2 (I think the new building system still uses gcc, right?)

Anyways, here's the relevant output from the compiler:

`FAILED: src/input_common/CMakeFiles/input_common.dir/gcadapter/gc_adapter.cpp.o 
/usr/bin/c++  -DARCHITECTURE_x86_64=1 -DBOOST_ALL_NO_LIB -DBOOST_CONTEXT_DYN_LINK -DBOOST_DATE_TIME_NO_LIB -DBOOST_ERROR_CODE_HEADER_ONLY -DBOOST_REGEX_NO_LIB -DBOOST_SYSTEM_NO_LIB -DFMT_LOCALE -DFMT_SHARED -DHAVE_SDL2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -I../src/. -I../externals/./microprofile -isystem /usr/include/SDL2 -O3 -DNDEBUG   -Wall -Werror=implicit-fallthrough -Werror=missing-declarations -Werror=reorder -Wextra -Wmissing-declarations -Wno-attributes -Wno-unused-parameter -mcx16 -std=gnu++17 -MD -MT src/input_common/CMakeFiles/input_common.dir/gcadapter/gc_adapter.cpp.o -MF src/input_common/CMakeFiles/input_common.dir/gcadapter/gc_adapter.cpp.o.d -o src/input_common/CMakeFiles/input_common.dir/gcadapter/gc_adapter.cpp.o -c ../src/input_common/gcadapter/gc_adapter.cpp
In file included from ../src/input_common/gcadapter/gc_adapter.cpp:8:
../src/./input_common/gcadapter/gc_adapter.h:11:10: fatal error: libusb.h: No such file or directory
   11 | #include <libusb.h>
      |          ^~~~~~~~~~
compilation terminated.`
This appears to be related to the recent gamecube controller support.
I don't know if any more information is required, so feel free to ask for any additional info

**Comment by lat9nq** [comment]
@jbeich someone beat me to it. I get that error on both my Manjaro 20 machine as well as my Arch Linux VM. This is caused by #4266.

**Comment by jbeich** [comment]
Can you show `LIBUSB_INCLUDE_DIR` value from CMakeCache.txt (in build directory) and where libusb.h is actually installed?

**Comment by lat9nq** [comment]
`LIBUSB_INCLUDE_DIR:PATH=/usr/include/libusb-1.0` which is a path that leads to libusb.h on my machine.

**Comment by ameerj** [comment]
![image](https://user-images.githubusercontent.com/52414509/87101823-1a0ba680-c21e-11ea-99c5-538cb51fe469.png)
This error is reported when building on windows. Commenting out `pkg_check_modules` in  externals/find-modules/FindLibUSB.cmake seems to have fixed the issue

**Comment by Shoegzer** [comment]
Confired this issue as well, and still encountering it after #4284 was merged.

**Comment by venomousZealot** [comment]
This helped me past the error:
`sudo ln -s /usr/include/libusb-1.0/libusb.h /usr/include/libusb.h`

#NotASolution

**Comment by stelford** [comment]
I faced the similar problem, but I also faced another issue where yuzu would start but then (almost immediately) hang on the loading screen. Of course, after a gdb and bt, it looks like it was getting hung up on the usb controller I had configured (my ps4 one using ds4drv). I basically recompiled and did a piecemeal of 'which libusb files are actually needed' and managed to figure out that ubuntu 20.04 installs the 'wrong' libusb-dev (at least wrong from yuzu's perspective). I ended up with the following;

```stef@stef-xubuntu:~/yuzu-mainline/build/bin$ dpkg -l | grep -i libusb
ii  libgusb2:amd64                                0.3.4-0.1                                   amd64        GLib wrapper around libusb1
ii  libusb-1.0-0:amd64                            2:1.0.23-2build1                            amd64        userspace USB programming library
ii  libusb-1.0-0:i386                             2:1.0.23-2build1                            i386         userspace USB programming library
ii  libusb-1.0-doc                                2:1.0.23-2build1                            all          documentation for userspace USB programming
ii  libusbmuxd6:amd64                             2.0.1-2                                     amd64        USB multiplexor daemon for iPhone and iPod Touch devices - library
ii  libusbredirhost1:amd64                        0.8.0-1                                     amd64        Implementing the usb-host (*) side of a usbredir connection (runtime)
ii  libusbredirparser1:amd64                      0.8.0-1                                     amd64        Parser for the usbredir protocol (runtime)```

**Comment by jbeich** [comment]
@stelford, I don't quite understand the cause but...
- [libusb ABI](https://abi-laboratory.pro/tracker/timeline/libusb/) is backward compatible
- yuzu-mainline builds are not linked against system libusb (not installed on CI by default)

**Comment by stelford** [comment]
It's strange, and I don't put it maybe past some package (davinci resolve or whatever) from installing needed dependencies but, I noticed that when I had libusb-1.0-0-dev, I would get past the compilation stage, but running would hang on the second 'h' spin in switch, every time, and sure enough, the bt inside gdb listed libusb as the point of conflict.. weird but.. as soon as I removed that, and made sure I had libusb installed, everything compiled *polite shrug* 

fwiw, I don't think you *can* get a linux system these days without libusb installed (ubuntu requires it for gnome-desktop in places) so, maybe try turning it on in the CI and see if anything explodes ?!?! jst a thought *tips hat*
